libraries/node-core-library/src/FileSystem.ts

```diff
@@ -521,4 +812,42 @@ export class FileSystem {
   public static getRealPath(linkPath: string): string {
     return fsx.realpathSync(linkPath);
   }
+
+  /**
+   * An async version of {@link FileSystem.getRealPath}.
+   */
+  public static getRealPathAsync(linkPath: string): Promise<string> {
+    return fsx.realpath(linkPath);
+  }
+
+  // ===============
+  // UTILITY FUNCTIONS
+  // ===============
+
+  /**
+   * Returns true if the error provided indicates the file or folder
+   * does not exist.
+   */
+  public static isNotExistError(error: NodeJS.ErrnoException): boolean {
+    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+  }
+
+  /**
+   * Re-throws the provided error if the error indicates the file or folder
+   * does not exist.
+   */
+  public static throwIfIsNotExistError(error: NodeJS.ErrnoException): void {
+    if (!FileSystem.isNotExistError(error)) {
+      throw error;
+    }
+  }
+
+  /**
+   * Resolves the provided paths relative to the provided parent path
+   * @param paths - The paths to resolve.
+   * @param relativeTo - The path relative to which the paths should be resolved.
+   */
+  private static _resolvePaths(paths: string[], relativeTo: string): string[] {
```

Comment:
I'm not sure that extracting this into a function actually makes the code easier to read, compared to simply copy+pasting the line. The reason is that `pathUtilities.resolve()` and `paths.map()` are fairly simple things that we already are familiar with, whereas `_resolvePaths()` requires us to look at the bottom of the file to figure out what is being resolved with respect to what.